src/main/java/org/broadinstitute/hellbender/utils/diffengine/DiffEngine.java

```diff
@@ -0,0 +1,369 @@
+package org.broadinstitute.hellbender.utils.diffengine;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.report.GATKReport;
+import org.broadinstitute.hellbender.utils.report.GATKReportTable;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.*;
+
+/**
+ * A generic engine for comparing tree-structured objects.
```

Comment:
Document the fact that this "generic engine " currently only supports two diffable formats (bam and vcf).